controllers/statusreport/statusreport_adapter.go

```diff
@@ -109,6 +110,17 @@ func (a *Adapter) EnsureSnapshotFinishedAllTests() (controller.OperationResult,
 		return controller.RequeueWithError(err)
 	}
 
+	// If for the snapshot there is an IntegrationTestScenario that is not triggered, it will add run labebl to snapshot
+	integrationTestScenarioNotTriggered := a.determineIfAllIntegrationTestsNotTriggered(integrationTestScenarios, testStatuses)
```

Comment:
We don't react to creation events here, so hopefully, it's enough to assume here, that this will only trigger new tests when status changed, and not randomly for any reconciliation for all old snapshots.

Just making note so others can check if I'm not mistaken.

Reply:
on the other hand, some defensive programming wouldn't hurt, if we change predicates in future